java/dataset/src/test/java/org/apache/arrow/dataset/TestDataset.java

```diff
@@ -56,15 +62,31 @@ protected RootAllocator rootAllocator() {
   protected List<ArrowRecordBatch> collectResultFromFactory(DatasetFactory factory, ScanOptions options) {
     final Dataset dataset = factory.finish();
     final Scanner scanner = dataset.newScan(options);
-    final List<ArrowRecordBatch> ret = stream(scanner.scan())
-        .flatMap(t -> stream(t.execute()))
-        .collect(Collectors.toList());
     try {
```

Comment:
Seems this should actually be a try-with-resources or a try-finally?

Reply:
I tried to factor out as try-finally but it seems that it better requires some common changes to AutoCloseables.java.

See my draft:
https://github.com/zhztheplayer/arrow-1/commit/c460f67bc893c8f1559c254965076358cd5332ee

Do you think we can have another ticket for the changes too?

Reply:
Ah, no worries then. Thanks for looking.